lib/workos/user_management.rb

```diff
@@ -746,6 +576,106 @@ def list_auth_factors(user_id:)
         )
       end
 
+      # Sends a verification email to the provided user.
+      #
+      # @param [String] id The unique ID of the User whose email address will be verified.
+      #
+      # @return WorkOS::UserResponse
+      sig do
+        params(
+          id: String,
+        ).returns(WorkOS::UserResponse)
+      end
+      def send_verification_email(id:)
+        response = execute_request(
+          request: post_request(
+            path: "/user_management/users/#{id}/email_verification/send",
+            auth: true,
+          ),
+        )
+
+        WorkOS::UserResponse.new(response.body)
+      end
+
+      # Verifies user email using one-time code that was sent to the user.
+      #
+      # @param [String] user_id The unique ID of the User whose email address will be verified.
+      # @param [String] code The one-time code emailed to the user.
+      #
+      # @return WorkOS::UserResponse
+      sig do
+        params(
+          user_id: String,
+          code: String,
+        ).returns(WorkOS::UserResponse)
+      end
+      def verify_email(user_id:, code:)
+        response = execute_request(
+          request: post_request(
+            path: "/user_management/users/#{user_id}/email_verification/confirm",
+            body: {
+              code: code,
+            },
+            auth: true,
+          ),
+        )
+        WorkOS::UserResponse.new(response.body)
+      end
+
+      # Resets user password using token that was sent to the user.
+      #
+      # @param [String] token The token that was sent to the user.
+      # @param [String] new_password The new password to set for the user.
+      #
+      # @return WorkOS::User
+      sig do
+        params(
+          token: String,
+          new_password: String,
+        ).returns(WorkOS::User)
+      end
+      def reset_password(token:, new_password:)
```

Comment:
FYI, those changes here are due to the re-ordering that I did, I'll raise further PRs to actually update the reset password methods.